src/datastructure/GlobalDescriptor.cpp

```diff
@@ -43,6 +43,20 @@ std::optional<GlobalDescriptor> GlobalDescriptor::build(GlobalDescriptorType typ
     return GlobalDescriptor(type, globalDescriptorToLengthType.at(type).second, buffer, globalDescriptorToLengthType.at(type).first);
 }
 
+SRef<GlobalDescriptor> GlobalDescriptor::buildSharedRef(GlobalDescriptorType type, unsigned char* buffer)
+{
+    if (!buffer) {
+        LOG_ERROR("Empty input data buffer, cannot build global descriptor");
+        return nullptr;
+    }
+    if (globalDescriptorToLengthType.find(type) == globalDescriptorToLengthType.end()) {
+        LOG_ERROR("Unsupported type, cannot build global descriptor");
+        return nullptr;
+    }
+    auto gd = GlobalDescriptor(type, globalDescriptorToLengthType.at(type).second, buffer, globalDescriptorToLengthType.at(type).first);
+    return xpcf::utils::make_shared<GlobalDescriptor>(gd);
```

Comment:
nit: you can check, but I think you can pass directly ctor args to `make_shared()` i.o. first creating the object, like:
```cpp
return xpcf::utils::make_shared<GlobalDescriptor>(type, globalDescriptorToLengthType.at(type).second, buffer, globalDescriptorToLengthType.at(type).first);
```

Reply:
At first, I did this way, but I had a compilation error related to xpcf. So I changed to the current implementation, everything builds well...

Reply:
Ah yes maybe due to the fact that the ctor is now private. Anyway, not a problem, approved.